### PR TITLE
optimize lookups of message by external id on status callbacks

### PR DESCRIPTION
**Pull request by nicpottier**
Turns out the planner shits the bed if you include the channel id. This just does the channel check after lookup.

### CODE REVIEW

**Comment by rowanseymour** [comment]
The other way to fix this (I think this would work) would be enforce uniqueness on external ids at the channel level - build a unique index on channel+external id. Tho we'd have to figure what to do when a dumb channel sends new messages with same eternal id.

**Comment by ericnewcomer** [comment]
External ids do get purged, so wouldn't be a huge conditional index, but we don't really need a new index with how nic solved it here, so that gets my vote.

File: temba/channels/handlers.py [hunk 1]
```diff
@@ -1380,9 +1380,9 @@ def post(self, request, *args, **kwargs):
             status = body['event_type']
 
             # look up the message
-            message = Msg.objects.filter(channel=channel, external_id=external_id).select_related('channel')
+            message = Msg.objects.filter(external_id=external_id).select_related('channel')
```
Comment (norkans7) [inline temba/channels/handlers.py]: .first() ??

Comment (norkans7) [reply temba/channels/handlers.py]: ohh we use `message` as queryset below then the if block needs `message[0].channel`

File: temba/channels/handlers.py [hunk 2]
```diff
@@ -2577,10 +2577,8 @@ def post(self, request, *args, **kwargs):
 
                     elif 'delivery' in envelope and 'mids' in envelope['delivery']:
                         for external_id in envelope['delivery']['mids']:
-                            msg = Msg.objects.filter(channel=channel,
-                                                     direction=OUTGOING,
-                                                     external_id=external_id).first()
-                            if msg:
+                            msg = Msg.objects.filter(external_id=external_id).first()
```
Comment (norkans7) [inline temba/channels/handlers.py]: select_related ?

**Comment by nicpottier** [comment]
I did a quick audit and I don't see any reason us just creating a conditional compound index of (channel_id, external_id) to replace just the external_id wouldn't be the best solution here. Which means I'll close this PR and open one with that instead. (I created that index already on RapidPro to avoid any potential melting and it is ~300M)

**Pull request closed without merging**